aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -637,10 +637,16 @@ void THTensor_(arange)(THTensor *r_, accreal xmin, accreal xmax, accreal step) {
   scalar_t i = 0;
 
   THArgCheck(step > 0 || step < 0, 3, "step must be nonzero");
+  THArgCheck(std::isfinite(static_cast<double>(xmin)) &&
+              std::isfinite(static_cast<double>(xmax))
+              , 1, "unsupported range: ", xmin, " -> ", xmax);
   THArgCheck(((step > 0) && (xmax >= xmin)) || ((step < 0) && (xmax <= xmin))
               , 2, "upper bound and larger bound inconsistent with step sign");
 
-  size = (ptrdiff_t) ceil((double)(xmax - xmin) / step);
+  double size_d = ceil(static_cast<double>(xmax - xmin) / step);
+  THArgCheck(size_d >= 0 && size_d <= static_cast<double>(PTRDIFF_MAX)
+             , 1, "invalid size, possible overflow?");
+  size = static_cast<ptrdiff_t>(size_d);
```

Comment:
Yeah, the `static_cast<double>(PTRDIFF_MAX)` gives me hives. I guess it's OK to not fix it properly for now, but if you don't get a comment here, all of the discussion on the PR will be lost to the sands fo time. Please put a comment here!

Reply:
@ezyang is there a better way though? as long as we're converting a double to an integral type there needs to be a cast, and at least now we've properly range checked it.

Reply:
I couldn't think of a better way to fix it, which is why I did not ask you to fix it :)

Reply:
FWIW this seems to be the boost way: https://www.boost.org/doc/libs/1_57_0/libs/numeric/conversion/doc/html/boost_numericconversion/improved_numeric_cast__.html (maybe its source code has a relatively self-contained function)